### DISCUSSION TITLE
Is there a tracing tool available for nextjs?

### DISCUSSION THREAD

**Discussion by akshitkrnagpal** [vercel/next.js#30515] [Help] [answered]
What I am trying to do is find out how much time does it take for each function to run. The idea is to find out which things are making the app slower.

**Comment by stigkj** [comment, marked as answer]
I guess you should be able to use any tracing tool that supports node.

**Reply by akshitkrnagpal** [reply]
Hi @stigkj 
Thanks for the reply.
Can you elaborate how? Let's assume I am trying to use Elastic APM https://www.elastic.co/guide/en/apm/agent/nodejs/current/custom-stack.html
Where should I exactly start this is nextjs app?
Which files comes before anything else in nextjs?

**Reply by stigkj** [reply]
I guess you could use the preload functionality of node, i.e. use the `--require` argument to load a js that starts the Elastic APM agent.
